src/MensaWidget/MWFrame.class.st

```diff
@@ -66,49 +70,138 @@ MWFrame class >> scrollBarThickness [
+MWFrame >> alignmentMorph [
+	
+	^ alignmentMorph
+]
 
-	| contentMorph |
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 15:03'
+}
+MWFrame >> alignmentMorph: anObject [
 	
-	contentMorph := ScrollPane new.
-	self fillContentMorph: contentMorph.
-	contentMorph scrollBarThickness: self class scrollBarThickness;
-		color: self class contentMorphColor;
+	alignmentMorph := anObject
+]
+
+{
+	#category : #creation,
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 20:55'
+}
+MWFrame >> buildContentMorph [
+
+	^ AlignmentMorph newColumn
+		cellInset: self class cellSpacing;
+		hResizing: #shrinkWrap;
+		vResizing: #shrinkWrap
+]
+
+{
+	#category : #creation,
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 21:04'
+}
+MWFrame >> buildLayout [
+
+	^ AlignmentMorph newColumn
+		hResizing: #spaceFill;
+		vResizing: #spaceFill;
 		position: self class contentPosition;
-		extent: self class contentExtent.
-	self addPaneMorph: contentMorph
+		name: 'Layout';
+		extent: self class contentExtent;
+		color: Color darkGray;
+		addMorph: self buildScrollPane;
+		addMorph: self buildMenuBar;
+		yourself
 ]
 
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'Louis Christ 6/7/2024 11:05'
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 21:02'
 }
-MWFrame >> fillContentMorph: aContentMorph [
+MWFrame >> buildMenuBar [
 	
+	^ self dayPicker: (MWMenuBar new target: self);
+		dayPicker.
+]
+
+{
+	#category : #creation,
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 20:55'
+}
+MWFrame >> buildScrollPane [
+
+	^ self contentMorph: (ScrollPane new 
+		scrollBarThickness: self class scrollBarThickness;
+		color: self class contentMorphColor;
+		extent: self class contentExtent;
+		yourself);
+		contentMorph
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 14:52'
+}
+MWFrame >> contentMorph [
+
+	^ contentMorph
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 14:52'
+}
+MWFrame >> contentMorph: anObject [
+
+	contentMorph := anObject
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 20:58'
+}
+MWFrame >> dayPicker [
+	^ dayPicker
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 20:58'
+}
+MWFrame >> dayPicker: anObject [
+	dayPicker := anObject
+]
+
+{
+	#category : #dayManipulation,
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 21:02'
+}
+MWFrame >> displayDay [
+
 	| contentMorph |
 	
-	contentMorph := AlignmentMorph newColumn
-		cellInset: self class cellSpacing;
-		hResizing: #shrinkWrap;
-		vResizing: #shrinkWrap.
-	MWDataGettingWrapper getTodaysMeals do: [:each | 
-		contentMorph addMorph: each getCanteenMorph].
-	aContentMorph scroller addMorph: contentMorph
+	self contentMorph scroller submorphsDo: [:each | each abandon].
+	contentMorph := self buildContentMorph.
+	(MWDataGettingWrapper getMealsOnDate: self dayPicker date yyyymmdd) do: 
+		[:each | contentMorph addMorph: each getCanteenMorph].
```

Comment:
regarding the each clauses, didn't we put the each on the line above and then do the linebreak?

Reply:
i would say if it is a oneliner we shouldn't

Reply:
I think line 162 is fine but I would agree Max regarding line 165-166

Reply:
to be honest, I can live with that policy to just not linebreak if it's a single line...

Reply:
it just looks cleaner and more readable and pretty print does it the same way